src/Microsoft.Performance.SDK.Runtime/CustomDataSourceReference.cs

```diff
@@ -61,36 +113,72 @@ protected CustomDataSourceReference(Type type, Func<ICustomDataSource> instanceF
         /// <exception cref="System.ObjectDisposedException">
         ///     <paramref name="other"/> is disposed.
         /// </exception>
-        protected CustomDataSourceReference(CustomDataSourceReference other)
+        public CustomDataSourceReference(CustomDataSourceReference other)
             : base(other.Type)
         {
+            this.dataSourceAttributes = other.dataSourceAttributes;
+
+            this.guid = other.guid;
+            this.name = other.name;
+            this.description = other.description;
+            this.commandLineOptionsRO = other.commandLineOptionsRO;
+
+            this.isDisposed = other.isDisposed;
```

Comment:
> this.isDisposed = other.isDisposed; [](start = 11, length = 36)

if other.isDisposed is true, we should never get this far and I'm not sure it makes sense to apply it to this instance anyway... since this instance's disposed state doesn't depend on other. this doesn't look right to me

Reply:
I agree; if we were using `other`'s public accessors it'd throw if it was disposed

Reply:
That is a good point, I'll have this just throw if other is disposed.